### DISCUSSION TITLE
User friendly error message when running into a delete constraint

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#12071] [Feature Requests]
When a user tries to delete an item and it's prevented by  a constraint (e.g deleting a related object) he get's a server error displayed.

I think a more specific error-message like "this item is used somewhere else and can't be deleted"  would improve the UX drastically. (MVP)

One step further the message could even tell the user where the element is in use (probably way to overcomplicated)

The current Error:

```
{
  "errors": [
    {
      "message": "An unexpected error occurred.",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

**Comment by rijkvanzanten** [comment]
Absolutely! We already have an abstraction in place for invalid foreign keys on saving, so we should also add one for deletion right here:

https://github.com/directus/directus/blob/main/api/src/exceptions/database/invalid-foreign-key.ts

Wanna take a stab at a PR? They can be a little finicky, as we have to do one for every supported vendor.

**Reply by Dominic-Marcelino** [reply]
Would really like to, but unfortunately I'm not enough into how Directus handles this part as well as other vendors 😅

**Reply by rijkvanzanten** [reply]
No prob! Like I mentioned, it's a finicky one, as we basically have to do some string matching magic to extract the exact error returned from each vendor, and sometimes different versions of the same database return errors slightly differently 😵

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
